File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& nums) {
        int min=INT_MAX,max=0;
        for(int i=0;i<nums.size();i++){
            min = std::min(min,nums[i]);
            max = std::max(max,nums[i]-min);
            // if(nums[i]<min){
                // min = nums[i];
            // }

            // if(nums[i]>max)max = std::max(max,max-min);
        }
        return max;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int low = Integer.MAX_VALUE;
        int high = Integer.MIN_VALUE;
        int ans = 0;

        for(int i=0;i<prices.length;i++){
            if(low>prices[i]){
                low = prices[i];
                high = prices[i];
            }
            if(high<prices[i]){
                high = prices[i];
            }
            ans = Math.max(ans,high-low);
        }
        return ans;
    }
}